src/CarbonAware.DataSources/CarbonAware.DataSources.Json/src/Configuration/JsonDataSourceConfiguration.cs

```diff
@@ -0,0 +1,52 @@
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace CarbonAware.DataSources.Json.Configuration;
+
+/// <summary>
+/// A configuration class for holding Json Data config values.
+/// </summary>
+public class JsonDataSourceConfiguration
+{
+    private const string BaseDir = "data-sources/json";
+    private const string DefaultDataFile = "test-data-azure-emissions.json";
+    private const string DirectoryRegExPattern = @"^[-/a-zA-Z_\d ]*$";
+    private string assemblyDirectory;
+    private string? dataFileLocation;
+
+    /// <summary>
+    /// Json data file location
+    /// </summary>
+    public string DataFileLocation
+    {
+        get => dataFileLocation!;
+        set
+        {
+            if (!IsValidDirPath(value))
+            {
+                throw new ArgumentException($"File path '{value}' contains not supported characters.");
+            }
+            dataFileLocation = Path.Combine(assemblyDirectory, BaseDir, value);
+        }
+    }
+
+    public const string Key = "JsonDataSourceConfiguration";
+
+    public JsonDataSourceConfiguration()
+    {
+        var assemblyPath = Assembly.GetExecutingAssembly().Location;
+        assemblyDirectory = Path.GetDirectoryName(assemblyPath)!;
+        DataFileLocation = DefaultDataFile;
+    }
+
+    private static bool IsValidDirPath(string fileName)
```

Comment:
Is there a check to see if the file exists?  By the name of the function, I was expecting a check done to ensure the file exists.  Also, I looked through the code in this PR for a 'File Does Not Exist' error message and could not find one.

Reply:
There is no check since https://github.com/microsoft/carbon-aware-sdk/blob/5c5f53db1e6068118d96b74a4b395e97ec4dc479/src/CarbonAware.DataSources/CarbonAware.DataSources.Json/src/JsonDataSource.cs#L127 takes care of it.

Reply:
:thumbsup: